tests/testthat/test-scalar_in_linter.R

```diff
@@ -0,0 +1,32 @@
+test_that("scalar_in_linter skips allowed usages", {
+  linter <- scalar_in_linter()
+
+  expect_lint("x %in% y", NULL, linter)
```

Comment:
Can we add a test for NA?

Reply:
Related: what's the parse data specifically for `NA_character_`?

Reply:
`<NUM_CONST>NA_character_</NUM_CONST>`, oddly enough, which is straight from the parser:

```r
getParseData(parse(text = "NA_character_"))
  line1 col1 line2 col2 id parent     token terminal          text
1     1    1     1   13  1      2 NUM_CONST     TRUE NA_character_
2     1    1     1   13  2      0      expr    FALSE              
```

Reply:
Test added.